roles/migrationcontroller/tasks/main.yml

```diff
@@ -5,6 +5,11 @@
 - block:
   - when: olm_managed
     block:
+
+    - k8s:
```

Comment:
I should probably check that this namespace already exists before applying this patch. The k8s yaml I'm including is incomplete and would probably fail to apply if the ns resource wasn't already present.

Reply:
I think in your template you just want to use `mig_namespace`. From defaults this is coming from the WATCH_NAMESPACE Env var that is populated from the operator namespace:

https://github.com/konveyor/mig-operator/blob/master/roles/migrationcontroller/defaults/main.yml#L24
`mig_namespace: "{{ lookup( 'env', 'WATCH_NAMESPACE') | default('openshift-migration') }}"`

Also if the state is present and it doesn't exist it will create it. I don't think there's missing information to create a namespace resource in your template?

```
$ oc get namespace foo
Error from server (NotFound): namespaces "foo" not found

$ cat test.yml monitoring-namespace-label.yml.j2 
- hosts: localhost
  gather_facts: false
  vars:
    mig_namespace: foo
  tasks:
  - k8s:
      state: "present"
      definition: "{{ lookup('template', 'monitoring-namespace-label.yml.j2') }}"
---
apiVersion: v1
kind: Namespace
metadata:
  labels:
    openshift.io/cluster-monitoring: "true"
  name: "{{ mig_namespace }}"

$ ansible-playbook test.yml 
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'

PLAY [localhost] ***************************************************************************************************************************************************************************

TASK [k8s] *********************************************************************************************************************************************************************************
changed: [localhost]

PLAY RECAP *********************************************************************************************************************************************************************************
localhost                  : ok=1    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   

$ oc get namespace foo
NAME   STATUS   AGE
foo    Active   51s
```

Reply:
Updated to use `mig_namespace`